File: backend/src/NutriSoftware.Domain/Entities/Alimento.cs
using NutriSoftware.Domain.Enums;

namespace NutriSoftware.Domain.Entities;

public class Alimento
{
    public Guid Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public FuenteAlimento Fuente { get; set; }
    public string? Categoria { get; set; }
    public decimal Energia { get; set; }
    public decimal Proteinas { get; set; }
    public decimal Grasas { get; set; }
    public decimal Carbohidratos { get; set; }
    public decimal Fibra { get; set; }
    public string? MicronutrientesJson { get; set; }
    public bool EsPersonalizado { get; set; }
    public Guid? NutricionistaId { get; set; }
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

    public Nutricionista? Nutricionista { get; set; }
}

File: frontend/src/types/alimento.ts
export type FuenteAlimento = 'TPCA' | 'SMAE' | 'USDA' | 'personalizado'

export type CategoriaAlimento =
  | 'Cereales y tubérculos'
  | 'Leguminosas'
  | 'Frutas'
  | 'Verduras'
  | 'Carnes y aves'
  | 'Pescados y mariscos'
  | 'Lácteos y huevos'
  | 'Grasas y aceites'
  | 'Azúcares'
  | 'Bebidas'
  | 'Varios'

export interface Alimento {
  id: string
  nombre: string
  fuente: FuenteAlimento
  categoria: CategoriaAlimento
  energia: number
  proteinas: number
  grasas: number
  carbohidratos: number
  fibra: number
  sodio?: number
  calcio?: number
  hierro?: number
  vitaminaC?: number
  esPersonalizado: boolean
  nutricionistaId?: string

  /**
   * Composición completa por 100 g, nutriente → valor, para los alimentos que
   * la traen (la tabla peruana aporta 22). Un nutriente en `null` significa
   * que la tabla no tiene el dato, no que valga cero.
   *
   * Opcional: las pantallas que solo usan macronutrientes la ignoran.
   */
  micronutrientes?: Record<string, number | null>
}

export interface PlatilloIngrediente {
  alimentoId: string
  alimentoNombre: string
  fuente: FuenteAlimento
  gramos: number
}

export interface NutrientesTotales {
  energia: number
  proteinas: number
  grasas: number
  carbohidratos: number
  fibra: number
}

export interface Platillo {
  id: string
  nombre: string
  descripcion?: string
  ingredientes: PlatilloIngrediente[]
  porciones: number
  creadoEn: string
}

export interface CrearAlimentoDto {
  nombre: string
  categoria: CategoriaAlimento
  energia: number
  proteinas: number
  grasas: number
  carbohidratos: number
  fibra: number
  sodio?: number
  calcio?: number
  hierro?: number
}

export interface CrearPlatilloDto {
  nombre: string
  descripcion?: string
  ingredientes: { alimentoId: string; gramos: number }[]
  porciones: number
}
